### ISSUE TITLE
Passing environment vars with  "_" in the name doesn't work

### ISSUE THREAD

**Issue by Technobeats** [inmanta/exec#4]
Defining an environment variable with "_" in results in a stack trace.

**WORKS**

JAVAHOME=

```
environment_vars = {"JAVAHOME": "/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/sbin:/sbin:/bin"}

exec::Run(host=target, command="ant deploy >> {{ _logfile }} 2>&1", environment=environment_vars, cwd=self.installation_path)
```

**ERROR**

JAVA_HOME=

```
environment_vars = {"JAVA_HOME": "/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/sbin:/sbin:/bin"}

exec::Run(host=target, command="ant deploy >> {{ _logfile }} 2>&1", environment=environment_vars, cwd=self.installation_path
```

Traceback (most recent call last):
  File "/opt/inmanta/lib64/python3.6/site-packages/inmanta/agent/handler.py", line 534, in execute
    self.do_changes(ctx, resource, changes)
  File "/var/lib/inmanta/60442c77-e62e-4fab-badd-350eed94b6fc/agent/code/modules/inmanta_plugins.exec.py", line 167, in do_changes
    return self.do_cmd(ctx, resource, resource.command)
  File "/var/lib/inmanta/60442c77-e62e-4fab-badd-350eed94b6fc/agent/code/modules/inmanta_plugins.exec.py", line 140, in do_cmd
    raise Exception("Failed to execute command: %s" % ret[1])
Exception: Failed to execute command:

**Comment by Technobeats** [comment]
I think the issue is not related to "_" in the var-name.

When I change JAVA_HOME to JAVA_1 the command is executed without errors.
 

**Works**
```
environment_vars = {
        "PATH" : "/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/sbin:/sbin:/bin"
    }
```

**Works**

```
environment_vars = {
        "JAVA_1" : "/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/sbin:/sbin:/bin"
    }
```

**Doesn't work**
```
environment_vars = {
        "JAVA_HOME" : "/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/sbin:/sbin:/bin"
    }
```

**Comment by wouterdb** [comment]
The actual problem is the '>>' 

The exec module only starts a shell when it needs to (i.e. when doing sudo).
So in most cases, operators like '>>' are not supported

To fix this, you can do the following

```
environment_vars = {"JAVAHOME": "/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/sbin:/sbin:/bin"}

exec::Run(host=target, command="sh -c 'ant deploy >> {{ _logfile }} 2>&1'", environment=environment_vars, cwd=self.installation_path)
```

**Comment by wouterdb** [comment]
fixed in 96fd533 v1.1.0
